desc/objectives/_omnigenity.py

```diff
@@ -519,6 +520,232 @@ def compute(self, params, constants=None):
+                rho=self._rho, theta=2 * field.M_well, zeta=8, NFP=field.NFP, sym=False
+            )
+        else:
+            grid_field = self._grid_field
+
+        self._dim_f = grid_field.num_nodes
+        self._eq_data_keys = ["|B|_mn"]
+        self._field_data_keys = ["|B|_omni", "h"]
+
+        assert grid_eq.NFP == grid_field.NFP
+        assert grid_eq.sym is False
+        assert grid_field.sym is False
+        assert grid_eq.num_rho == 1
+        assert grid_field.num_rho == 1
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        self._profiles = get_profiles(self._eq_data_keys, obj=eq, grid=grid_eq)
+        self._eq_transforms = get_transforms(
+            self._eq_data_keys,
+            obj=eq,
+            grid=grid_eq,
+            M_booz=M_booz,
+            N_booz=N_booz,
+        )
+        self._field_transforms = get_transforms(
+            self._field_data_keys,
+            obj=field,
+            grid=grid_field,
+        )
+        self._constants = {
+            "equil_transforms": self._eq_transforms,
+            "equil_profiles": self._profiles,
+            "field_transforms": self._field_transforms,
+            "helicity": self.helicity,
+        }
+
+        timer.stop("Precomputing transforms")
+        if verbose > 1:
+            timer.disp("Precomputing transforms")
+
+        if self._normalize:
+            # average |B| on axis
+            self._normalization = jnp.mean(field.B_lm[: field.M_well])
+
+        super().build(use_jit=use_jit, verbose=verbose)
+
+    def compute(self, equil_params, field_params, constants=None):
+        """Compute omnigenity errors.
+
+        Parameters
+        ----------
+        equil_params : dict
+            Dictionary of equilibrium degrees of freedom, eg Equilibrium.params_dict
+        field_params : dict
+            Dictionary of field degrees of freedom, eg OmnigenousField.params_dict
+        constants : dict
+            Dictionary of constant data, eg transforms, profiles etc. Defaults to
+            self.constants
+
+        Returns
+        -------
+        f : ndarray
+            Omnigenity error at each node (T).
+
+        """
+        if constants is None:
+            constants = self.constants
+        eq_data = compute_fun(
+            "desc.equilibrium.equilibrium.Equilibrium",
+            self._eq_data_keys,
+            params=equil_params,
+            transforms=constants["equil_transforms"],
+            profiles=constants["equil_profiles"],
+        )
+        field_data = compute_fun(
+            "desc.magnetic_fields.OmnigenousField",
+            self._field_data_keys,
+            params=field_params,
+            transforms=constants["field_transforms"],
+            profiles={},
+        )
+
+        # TODO: move to a compute function?
+        # would need to add iota as an OmnigenousField attribute
```

Comment:
could also make iota an extra kwarg to the compute function (like `gamma` or `helicity`)? Then you could just copy/paste this and pass in the value of iota from the equilibrium